src/main/java/org/cryptomator/ui/error/ErrorController.java

```diff
@@ -42,7 +42,8 @@ public class ErrorController implements FxController {
 
 	private static final ObjectMapper JSON = new ObjectMapper();
 	private static final Logger LOG = LoggerFactory.getLogger(ErrorController.class);
-	private static final String ERROR_CODES_URL = "https://api.cryptomator.org/desktop/error-codes.json";
+	private static final String USER_AGENT_VERSION_FORMAT = "Cryptomator/%s (Build %s)";
```

Comment:
Adding the build number is a good idea, but let's add the OS/arch as well. This is the user agent string that our update checker uses:

https://github.com/cryptomator/cryptomator/blob/bcb970afb1639fe5869a3670fbb9d2eeb8e0d2c6/src/main/java/org/cryptomator/ui/fxapp/UpdateCheckerModule.java#L57-L67

In a separate PR, we can dedup this and use the same user agent across all places.

Reply:
I looked with mindmonk at common user agents. According to the RFCc, the user agent can have several comments in a row and since OS related architecture is [normally](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent) put in the comment, we decided to use the format `Cryptomator/<version> (Build <flavor>) (<OS><version><arch>)`

Changed in https://github.com/cryptomator/cryptomator/pull/3109/commits/830970cb75726db8b484e834743cd62cff6f32cc